src/Lecture3.hs

```diff
@@ -179,9 +229,9 @@ together only different elements.
 [1,2,3,4,5]
 >>> appendDiff3 (Product 2) (Product 3) (Product 3)
 Product {getProduct = 6}
-
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (Eq a, Monoid a) => a -> a -> a -> a
+appendDiff3 a b c = mconcat $ nub [a, b, c]
```

Comment:
That's a clever and concise implementation! 👏🏻 

However, the type signature is more strict than required. Your function works with `Monoid`. But not every type can have a `Monoid` instance.

This means, that `appendDiff3` can't work with types like `List1`. But if you change the constraint from `Monoid` to `Semigroup`, this function should still work. Though, relaxing the constraint will require changing the implementation as well 🚧